docs/references/example_source.py

```diff
@@ -0,0 +1,30 @@
+class ExampleSourceContext(BaseSourceContext):
+    async def update(self, record):
+        self.parent.mem[record.key] = record
+
+    async def records(self) -> AsyncIterator[Record]:
+        for record in self.parent.mem.values():
+            yield record
+
+    async def record(self, key: str) -> Record:
+        return self.parent.mem.get(key, Record(key))
+
+
+@config
+class ExampleSourceConfig:
+    records: List[Record]
+
+
+class ExampleSource(BaseSource):
```

Comment:
In this case, you've duplicated `MemorySource`, therefore it's probably be we just use `MemorySource` throughout the docs. We talked about making `Example<PluginType>` classes, however with this one we probably don't need to do that since there's already `MemorySource`.

I realized that we can do the same thing with the model examples. When you go to do the model, let's make sure https://github.com/intel/dffml/issues/500 gets done first, and then you could use that within the docstrings. This way you don't have to make an `Example` class for it.

Reply:
Well yes, it was bugging me too. 
Sure thing! Thank you.